# Worksheet #4

More on arrays.

Let's play with the data we were using in class on Friday.

First our usual boilerplate.

In [2]:
%matplotlib inline
import numpy as np
import pylab

This next cell will grab the data from the web.

In [5]:
babies = np.genfromtxt('http://goo.gl/BRgpl1')

# 1. How many babies are there in these data?

Hint: Try using the **len** function. 

In [3]:
len(babies)
babies.shape

(439, 2)

I got 439.

# 2. What is the average weight of the babies?

Be careful to split out the columns or otherwise be sure you get the mean of only the weights. Just saying `np.mean(babies)` won't do the job.

In [4]:
np.mean(babies[:,0])

5.4800227790432805

I got a tiny bit over 5.48 pounds.

# 3. How many of the babies are girls?

In [5]:
np.count_nonzero(babies[:,1])

210

I got 210.

# 4. How many are boys?

In [7]:
np.sum(babies[:,1]==0)

229

I got 229.

# 5. How much heavier on average are the girls than the boys?

In [15]:
np.mean(babies[babies[:,1]==1,0])-np.mean(babies[babies[:,1]==0,0])

0.045782491162404071

I got about 0.046 pounds heavier. 

You can submit your worksheet using the cell below.

In [24]:
# do not change the code below
import comp116
comp116.submit('ws4')